### ISSUE TITLE
Add model import and export functionality

### ISSUE THREAD

**Issue by florian-huber** [matchms/ms2deepscore#34]
The `Model` class we now have contains a `.base` and a head model (`.model`). In addition we most likely will add the used `SpectrumBinner` as well. So we need to add methods for import and export.
e.g.
```python
model.save(filename)
model.load(filename)
```
depending on what file format we are going to use. Could be `hdf5` or -at least for now- just pickled...

**Comment by florian-huber** [comment]
I will probably start by something like this first:
```python
#%%
import json
from tensorflow.python.keras.saving import hdf5_format
import h5py

filename = "siamese.hdf5"

binner_dict = ms2ds_binner.__dict__
binner_json = json.dumps(binner_dict)

# Save model
with h5py.File(filename, mode='w') as f:
    hdf5_format.save_model_to_hdf5(model.model, f)
    f.attrs['spectrum_binner'] = binner_json

# Load model
with h5py.File(filename, mode='r') as f:
    param1 = f.attrs['param1']
    keras_model = hdf5_format.load_model_from_hdf5(f)
```

**Comment by florian-huber** [comment]
Export and import to and from single hdf5 file was implemented in  #36.
